SYCL/ESIMD/api/functional/ctors/ctor_move.cpp

```diff
@@ -0,0 +1,233 @@
+//==------- ctor_move.cpp  - DPC++ ESIMD on-device test --------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu, level_zero
+// XREQUIRES: gpu
+// TODO Remove the level_zero restriction once the test is supported on other
+// platforms
+// UNSUPPORTED: cuda, hip
+// XRUN: %clangxx -fsycl %s -fsycl-device-code-split=per_kernel -o %t.out
```

Comment:
@vladimirlaz - what is BKM for such tests. Should they be compiled/run and report fail, or skip the test compilation/run completely?
My concern regarding no-compilation and no-run approach used above is that it is unclear if the test is good, if it is compilable and does not have syntax errors.

Reply:
Fair point. Please take a look into the 37e5e50466269c3e6e913393cb41b305915a1bbf as a suggestion.
The other option is to use `RUN: %clangxx -DSKIP_VECTOR_LEN_32 -fsycl ...`. By doing so we would be able:

- to disable some checks in CI
- to enable all checks "by default", making it possible to switch them off one-by-one in case of multiple feature flags during debugging session - with no changes to the test file itself

Still I'm not sure if the `RUN: %clangxx -DSKIP_VECTOR_LEN_32` solution would work for every setup or use-case, therefore the hard-coded macro was used.
Any feedback appreciated.